rtt/internal/ConnectionIntrospector.cpp

```diff
@@ -235,71 +234,95 @@ ConnectionIntrospector::NodePtr ConnectionIntrospector::findNode(const Node &nod
+ConnectionIntrospector::ConnectionPtr ConnectionIntrospector::findConnectionTo(const Connections &connections, const Node &node) {
     Connections::const_iterator it;
     for(it = connections.begin(); it != connections.end(); ++it) {
         if (*((*it)->to()) == node) return *it;
     }
     return ConnectionPtr();
 }
 
-//static std::ostream& ConnectionIntrospector::printIndented(std::ostream& os,
-//        int i) const {
-//    const int currIndent = 4 * this->depth + i;
-//    if (this->depth == -1) {
-//        // For depth -1, only log the component name.
-//        os << std::string(i, ' ') << this->in_port.owner_name
-//           << " COMPONENT\n";
-//    } else if (this->depth == 0) {
-//        // For depth 0, only log the port.
-//        std::string connection_summary;
-//        const int connection_nr = this->sub_connections.size();
-//        switch (connection_nr) {
-//        case 0:
-//            connection_summary = "No";
-//            break;
-//        case 1:
-//            connection_summary = "Single";
-//            break;
-//        default:
-//            connection_summary = "Multiple";
-//            break;
-//        }
-//        os << std::string(currIndent, ' ')
-//           << (this->is_forward ? " IN " : " OUT ")
-//           << (this->is_forward ? this->in_port
-//                                     : this->out_port)
-//           << " with " << connection_summary << " connection(s) (#"
-//           << connection_nr << ")" << ":\n";
-//    } else if (this->depth > 0) {
-//        // Positive depth, log the full connection information.
-//        os << std::string(currIndent, ' ')
-//           << (this->is_forward ? " -> IN " : " <- OUT ")
-//           << ((this->is_forward ? this->in_port
-//                                     : this->out_port).is_remote ?
-//                  "(remote) " : "")
-//           << (this->is_forward ? this->in_port
-//                                     : this->out_port)
-//           << " : (" << this->connection_policy << ")\n";
-//    }
-//    for (std::list< ConnectionIntrospector >::const_iterator
-//            it = this->sub_connections.begin();
-//         it != this->sub_connections.end(); ++it) {
-//        os << *it;
-//    }
-//    return os;
-//}
-
-//boost::function<std::ostream&(std::ostream&)>
-//ConnectionIntrospector::indent(int i) const {
-//    return boost::bind(&ConnectionIntrospector::printIndented, this, _1, i);
-//}
+std::string ConnectionIntrospector::Node::getConnectionSummary() const {
+    std::string connection_summary;
+    const int connection_nr = this->connections_.size();
+    switch (connection_nr) {
+    case 0:
+        connection_summary = "no";
+        break;
+    case 1:
+        connection_summary = "single";
+        break;
+    default:
+        connection_summary = "multiple";
+        break;
+    }
+
+    return (" with " + connection_summary + " connection(s) (#"
+            + std::to_string(connection_nr) + ")");
```

Comment:
`std::to_string` is C++11 only. You can use boost::lexical_cast<std::string>` instead.

Reply:
Fixed, thanks :+1: